### DISCUSSION TITLE
How to retreive users' avatar in an interface?

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#14486] [Archive]
Hi,
In a rich read-only interface, I want to retrieve multiple avatars of some users.
In an M2M, the avatar HTLM generataed by Directus looks like
`<img src="/assets/1176c517-64b4-4df6-b248-0e8a92f3374c?key=system-small-cover&access_token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjA0YzY2Y2M1LTFhMTgtNGFmNi04ZGYyLWJiNWQwYTM5NGM0MSIsInJvbGUiOiJlMDI2NGU0Ny0yNWQxLTQzYzEtOWI3MS0wZGMyUWU4OGMzMzQiLCJhcHBfYWNjZXNzIjp0cnVlLCJhZG1pbl9hY2Nlc3MiOnRydWUsImlhdCI6MTY1ODEzODU2MSwiZXhwIjoxNjU4MTM5NDYxLCJpc3MiOiJkaXJlY3R1cyJ9.LvmNbI9oYRvFA3wwEmwH6F8p3oaETPI5UDWpfMesRXE" etc. >`

The `access_token` is different for every avatar.

Is there a way to set the correct `access_token` query string for each avatar in the component?

**Comment by rijkvanzanten** [comment]
The `api` object injected through the `useApi()` composable that's exported in the extensions-sdk has the access token in it's default headers. Any request made through the `api` object there will be authenticated with the correct token. If you need access to the access token as a string, you can use a similar flow as: https://github.com/directus/directus/blob/e6e3b597fef2538be477a369309095980cedb9f9/app/src/api.ts#L106-L115
